enhancements/cloud-integration/openstack/failure-domains.md

```diff
@@ -0,0 +1,715 @@
+          fixedIPs:
+          - subnetID: 61d376b0-3714-4549-b017-3a5e0007f07c
+          tags:
+          - "control-plane"
+  replicas: 3
+compute:
+- name: worker
+  platform:
+    openstack:
+      type: m1.large
+      rootVolume:
+        size: 50
+        type: silver
+      failureDomains:
+      - computeAvailabilityZone: az1
+        storageAvailabilityZone: az1
+        ports:
+        - networkID: 4a270cf8-f5f9-4a09-8cfc-d39d77aa2ba1
+          fixedIPs:
+          - subnetID: 31ad4088-7e52-48da-9df5-35e70da438f6
+          tags:
+          - "control-plane"
+      - computeAvailabilityZone: az2
+        storageAvailabilityZone: az2
+        ports:
+        - networkID: dbc8664c-feee-4168-854b-75f8eb8eb258
+          fixedIPs:
+          - subnetID: c23f88c0-910a-4d4a-9ca6-9a11a0d71f40
+          tags:
+          - "control-plane"
+      - computeAvailabilityZone: az3
+        storageAvailabilityZone: az3
+        ports:
+        - networkID: a996c932-2aa1-4095-a40e-765bf844fcb4
+          fixedIPs:
+          - subnetID: 61d376b0-3714-4549-b017-3a5e0007f07c
+          tags:
+          - "control-plane"
+  replicas: 6
+```
+
+Note that by specifying `machinesSubnet` the installer will not create a default control plane network. Any future workers created without explicit ports will be attached to this subnet.
+
+The created Machines will contain the relevant failure domain in full in their `OpenStackProviderSpec`, e.g.:
+
+```yaml
+  ...
+  flavor: m1.large
+  rootVolume:
+    volumeType: gold
+    size: 50
+  failureDomain:
+    computeAvailabilityZone: az1
+    storageAvailabilityZone: az1
+    ports:
+    - networkID: 4a270cf8-f5f9-4a09-8cfc-d39d77aa2ba1
+      fixedIPs:
+      - subnetID: 31ad4088-7e52-48da-9df5-35e70da438f6
+      tags:
+      - "control-plane"
+```
+
+#### Upgrade an existing cluster to use control plane failure domains
+
+This is not supported until we integrate with Control Plane MachineSet.
+
+#### Upgrade an existing cluster to use worker failure domains
+
+Note that there is currently little reason to do this. The only functionality we currently have for distributing workers across failure domains is in the installer, so not relevent to an existing cluster. However, the user would do this by creating a new MachineSet with a template containing the failure domain definition. e.g.:
+
+```yaml
+spec:
+  ...
+  replicas: 5
+  template:
+    ...
+    spec:
+      providerSpec:
+        value:
+          apiVersion: openstackproviderconfig.openshift.io/v1alpha1
+          kind: OpenstackProviderSpec
+          ...
+          flavor: m1.large
+          image: rhcos
+          failureDomain:
+            computeAvailabilityZone: az1
+            ports:
+            - networkID: 4a270cf8-f5f9-4a09-8cfc-d39d77aa2ba1
+              fixedIPs:
+              - subnetID: 31ad4088-7e52-48da-9df5-35e70da438f6
+              tags:
+              - "control-plane"
+```
+
+#### Variation [optional]
```

Comment:
Should two failureDomains also be supported?

Reply:
No. One `MachineSet` defines a set of identical machines deployed in the same failure domain. If the user wants to spread machines across failure domains, they will have to create one `MachineSet` per failure domain.